### PR TITLE
Using string.match not regexp.test

### PR DESCRIPTION
**Pull request by kfbishop**
[My first PR, please forgive any failings]
In the demo, I sorted the cities array such that the RegExp.test() shows the failure, try searching on "Ma" will result in Madrid showing, Manchester not showing.

Ref Issue: https://github.com/valor-software/ng2-select/issues/56

### CODE REVIEW

File: demo/components/select/single-demo.ts [hunk 1]
```diff
@@ -16,15 +16,7 @@ export class SingleDemo {
...
-    'Essen', 'Frankfurt', 'Genoa', 'Glasgow', 'Gothenburg', 'Hamburg', 'Hannover',
-    'Helsinki', 'Leeds', 'Leipzig', 'Lisbon', 'Łódź', 'London', 'Kraków', 'Madrid',
-    'Málaga', 'Manchester', 'Marseille', 'Milan', 'Munich', 'Naples', 'Palermo',
-    'Paris', 'Poznań', 'Prague', 'Riga', 'Rome', 'Rotterdam', 'Seville', 'Sheffield',
-    'Sofia', 'Stockholm', 'Stuttgart', 'The Hague', 'Turin', 'Valencia', 'Vienna',
-    'Vilnius', 'Warsaw', 'Wrocław', 'Zagreb', 'Zaragoza'];
+  private items:Array<string> = ['Amsterdam', 'Antwerp', 'Athens', 'Barcelona', 'Berlin', 'Birmingham', 'Bradford', 'Bremen', 'Brussels', 'Bucharest', 'Budapest', 'Cologne', 'Copenhagen', 'Dortmund', 'Dresden', 'Dublin', 'Düsseldorf', 'Essen', 'Frankfurt', 'Genoa', 'Glasgow', 'Gothenburg', 'Hamburg', 'Hannover', 'Helsinki', 'Kraków', 'Leeds', 'Leipzig', 'Lisbon', 'London', 'Madrid', 'Manchester', 'Marseille', 'Milan', 'Munich', 'Málaga', 'Naples', 'Palermo', 'Paris', 'Poznań', 'Prague', 'Riga', 'Rome', 'Rotterdam', 'Seville', 'Sheffield', 'Sofia', 'Stockholm', 'Stuttgart', 'The Hague', 'Turin', 'Valencia', 'Vienna', 'Vilnius', 'Warsaw', 'Wrocław', 'Zagreb', 'Zaragoza', 'Łódź'];
```
Comment (binarious) [inline demo/components/select/single-demo.ts]: Why this change? It was better before.

Comment (kfbishop) [inline demo/components/select/single-demo.ts]: Look at the defects addressed.   And what does "better" mean to you?

Comment (binarious) [inline demo/components/select/single-demo.ts]: What defects do you mean? Your version exceeds editor's line length.

Comment (kfbishop) [inline demo/components/select/single-demo.ts]: https://github.com/valor-software/ng2-select/issues/56
and
https://github.com/valor-software/ng2-select/issues/54

Each has an exemplar plunker for review...

Comment (binarious) [inline demo/components/select/single-demo.ts]: I was just asking why you changed the coding style from multiple lines to a single line.

Comment (kfbishop) [inline demo/components/select/single-demo.ts]: My intention was to just sort the list to better expose the filtering defect.  I could/should have put in line breaks.

**Comment by kfbishop** [comment]
There's still issues here with the rich options.   The highlight pipe can break it, and also the <input> value wont show HTML, so get raw markup once selected.  Might want to hold off on this PR for now ...   Thoughts and ideas welcomed though.

**Comment by valorkin** [comment]
@kfbishop first of all thanks for your time and efforts! :)
I will check it asap,
if you will squash and force push commits to your branch it will help a lot
with commit name
`fix(select): invalid filtering due to usage of regexp.test. fixes #56`
thanks in advance

**Comment by kfbishop** [comment]
Did 2 squashes for #54 and #56. Sorry for the combined PR.  Will use branches next time.
